### DISCUSSION TITLE
Unable to mount/import zfs drive. Uberblock available

### DISCUSSION THREAD

**Discussion by theDepart3d** [openzfs/zfs#15822] [Q&A] [answered]
Hi All,
I have a single drive ZFS drive named "MV-Storage-3TB". I basically switched hardware and a clean proxmox install.
I am now unable to import the pool.

```
root@node:~# zpool import
   pool: MV-Storage-3TB
     id: 11250951926662925056
  state: ONLINE
status: The pool was last accessed by another system.
 action: The pool can be imported using its name or numeric identifier and
	the '-f' flag.
   see: https://openzfs.github.io/openzfs-docs/msg/ZFS-8000-EY
 config:

	MV-Storage-3TB                              ONLINE
	  ata-WDC_WD30EZRX-00MMMB0_WD-WCAWZ2611164  ONLINE
root@node:~# zpool import -f MV-Storage-3TB
cannot import 'MV-Storage-3TB': one or more devices is currently unavailable
```

I then ran the following command: zdb -e MV-Storage-3TB
The results are quite large and i have attached the full output [zdb_e_resutls.txt](https://github.com/openzfs/zfs/files/14056668/zdb_e_resutls.txt)

in the results there is the following which i 100% do not understand. 
This is the very last output from the command:
```
ZFS_DBGMSG(zdb) START:
spa.c:6110:spa_import(): spa_import: importing MV-Storage-3TB
spa_misc.c:418:spa_load_note(): spa_load(MV-Storage-3TB, config trusted): LOADING
vdev.c:160:vdev_dbgmsg(): disk vdev '/dev/disk/by-id/ata-WDC_WD30EZRX-00MMMB0_WD-WCAWZ2611164-part1': best uberblock found for spa MV-Storage-3TB. txg 2756441
spa_misc.c:418:spa_load_note(): spa_load(MV-Storage-3TB, config untrusted): using uberblock with txg=2756441
spa.c:8392:spa_async_request(): spa=MV-Storage-3TB async request task=2048
spa_misc.c:418:spa_load_note(): spa_load(MV-Storage-3TB, config trusted): LOADED
spa.c:8392:spa_async_request(): spa=MV-Storage-3TB async request task=32
spa.c:8392:spa_async_request(): spa=MV-Storage-3TB async request task=4
spa.c:8392:spa_async_request(): spa=MV-Storage-3TB async request task=4
ZFS_DBGMSG(zdb) END
```
Most of the output from the above, i have no clue. I simply cannot figure out how to import or repair this pool.

Any assistance or help would be greatly appreciated.

**Comment by rincebrain** [comment]
You'd be best off probably looking at what's in `/proc/spl/kstat/zfs/dbgmsg` after trying an import, since zdb doesn't seem to have any complaints.

**Reply by theDepart3d** [reply]
Thank you, I am trying the following
```$zpool import -f -T 2756441 -F MV-Storage-3TB```
This has been running for 2hours without a single output to terminal.

The ```cat /proc/spl/kstat/zfs/dbgmsg``` has the following for the above command

"spa_misc.c:418:spa_load_note(): spa_load(MV-Storage-3TB, config trusted): performing a complete scan of the pool since extreme rewind is on. This may take a very long time."

Yesterday it took 10 Hours for the zdb -e command. So i guess ill have to wait for a good couple of hours for this to complete and check the results.

**Reply by theDepart3d** [reply, marked as answer]
`zpool import -f -T 2756441 -F MV-Storage-3TB`

the import worked. took around 13 hours though. Will be testing it today to check if the data has been corrupted or not.
